rfcs/01924_tls_attestation_ca.md

```diff
@@ -0,0 +1,169 @@
+   1. When requesting a TEE report from the TEE firmware, **Server** also puts
+      the hash of its TLS public key into it
+   1. For example, remote attestation could be done via
+      [#1929](https://github.com/project-oak/oak/pull/1929)
+1. **Server** sends **CA** a certificate signing request
+   1. Communication happens over a secure channel creates as part of the remote
+      attestation
+1. **CA** creates a signed certificate and sends it back to the **Server**
+   1. The certificate contains **Server**'s TEE report
+
+<!-- From: -->
+<!-- https://sequencediagram.googleplex.com/view/6235412701904896 -->
+<img src="images/tls_attestation_ca_workflow_server.png" width="850">
+
+Second stage corresponds to the client connection:
+
+1. [_Optional_] **Client** remotely attests the **CA**
+1. **CA** sends **Client** its root certificate
+   1. **Client** trusts that this is a correct certificate, since it was sent
+      via a secure channel created during the attestation process
+   1. **Client** needs to request the root certificate only once at startup, and
+      then the client can use it for all **Server** connections
+1. **Client** connects to the **Server** using TLS
+1. **Server** sends **Client** a TLS certificate previously signed by **CA**
+
+1. **Client** checks the validity of the **Server**’s TLS certificate
+   1. **Client** verifies that the TLS certificate provided by the **Server**
+      was signed by the **CA**’s root certificate
+   1. **Client** also checks the TEE report provided in the **Server**’s TLS
+      certificate
+      1. It checks that the TEE report was generated using the genuine TEE
+         firmware key
+      1. It checks that the TEE report contains the hash of the **Server**’s
+         public key
+   1. If the **Server**’s TLS certificate is not valid, then the **Client**
+      closes the connection and aborts the protocol
+1. Since **Client** trusts **CA** (and has its root certificate) it can now
+   trust **Server** and establish a secure connection
+
+<!-- From: -->
+<!-- https://sequencediagram.googleplex.com/view/6490702202535936 -->
+<img src="images/tls_attestation_ca_workflow_client.png" width="850">
+
+Several important notes:
+
+- Client needs to **trust** in the CA and Server implementation correctness:
+  - Main requirements are:
+    - CA should correctly attest the Server and sign right certificates
+    - Server shouldn’t leak its TLS keys
+  - Client can gain trust by accessing the code of the CA and the Server, and
+    also by **reproducibly building** binaries on their machine
+    - Though it could be done by a third party reviewer that the Client can
+      trust
+- Client only needs to **attest the CA once** and then use its root certificate
+  for multiple connections to any Server
+- Remote attestation logic on Servers can be implemented as a **separate
+  binary**, that attests to the CA, receives a signed certificate and puts it as
+  an input to the TEE application
```

Comment:
I don't quite understand this sentence. What kind of input is this? It sounds like the application still needs to handle this input, which to me sounds like a remote-attestation specific logic.

Reply:
The input here is the certificate.
Since the application used TLS already, it needed a certificate.
We are just changing this certificate to the one created by the Attestation CA.